packages/salesforce/test/Adaptor.test.js

```diff
@@ -194,4 +195,138 @@ describe('Adaptor', () => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+
+      let state = {
+        connection: fakeConnection,
+        references: [],
+        qs: 'select Name from Account',
+      };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query(state => state.qs)(state).then(state => {
+        expect(spy.calledWith('select Name from Account')).to.eql(true);
+      });
+    });
+
+    it('should fetch up to 2000 records', done => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+      let state = { connection: fakeConnection, references: [] };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query('select Name from Account')(state)
+        .then(state => {
+          expect(spy.called).to.eql(true);
+          expect(state.references[0]).to.eql({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        })
+        .then(done);
+    });
+    it('should fetch all page if autoFetch is true', done => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: false,
+            totalSize: 5713,
+            nextRecordsUrl:
+              '/services/data/v47.0/query/0r8yy3Dlrs3Ol9EACO-2000',
+            records: [{ Name: 'Open' }],
+          });
+        },
+        request: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 5713,
+            records: [{ Name: 'Fn' }],
+          });
+        },
+      };
+      let state = { connection: fakeConnection, references: [] };
+      let spy = sinon.spy(fakeConnection, 'query');
+      let spyReq = sinon.spy(fakeConnection, 'request');
+
+      query('select Name from Account', { autoFetch: true })(state)
+        .then(state => {
+          expect(spy.called).to.eql(true);
+          expect(spyReq.called).to.eql(true);
+          expect(state.references[0]).to.eql({
+            done: true,
+            totalSize: 5713,
+            records: [{ Name: 'Open' }, { Name: 'Fn' }],
+          });
+        })
+        .then(done);
+    });
+    it('should not fetch another page if autofetch is false', done => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: false,
+            totalSize: 5713,
+            nextRecordsUrl:
+              '/services/data/v47.0/query/0r8yy3Dlrs3Ol9EACO-2000',
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+      let state = { connection: fakeConnection, references: [] };
+      let spy = sinon.spy(fakeConnection, 'query');
```

Comment:
Just to really drive this test home, can we add a mock `request` function (which does nothing) and assert that it isn't called?

And in fact, we don't need to assert that `query` is called at all. We know it's called because we're checking the result data (and anyway we tested that earlier)

If I set `autofetch` to true the test throws an exception, which is good because it means  it's working as designed. But a better assertion will give me confidence that the test works without having to run (and modify) the code